lib/ddtrace/configuration.rb

```diff
@@ -103,6 +103,7 @@ def reset!
       safely_synchronize do |write_components|
         @components.shutdown! if components?
         write_components.call(nil)
+        configuration.reset!
       end
     end
```

Comment:
> This currently only affects our test suite, as Datadog.reset! is currently only invoked from our spec helper.

Since this operation is quite error-prone, what do you think of either marking `reset!` as private, or just moving it to a spec helper?

I'm concerned that `reset!` looks like something that we don't ever customers to use, so we should make it a bit more clearer by removing/hiding it.

Reply:
I think it should have always been private. Changing it.